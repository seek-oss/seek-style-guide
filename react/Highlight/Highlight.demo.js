import React from 'react';
import Highlight from './Highlight';
import Text from '../Text/Text';
import { TONE } from '../private/tone';

const renderChildren = props => [
  'This text is ',
  <Highlight key="highlight" {...props}>
    highlighted
  </Highlight>
];

export default {
  route: '/highlight',
  title: 'Highlight',
  category: 'Typography',
  component: Text,
  initialProps: {
    headline: true,
    children: renderChildren({})
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Secondary text',
          transformProps: props => ({
            ...props,
            secondary: true,
            children: renderChildren({ secondary: true })
          })
        }
      ]
    },
    {
      label: 'Tone',
      type: 'radio',
      states: [
        {
          label: 'Select tone...'
        },
        {
          label: 'Critical',
          transformProps: props => ({
            ...props,
            children: renderChildren({
              secondary: props.secondary,
              tone: TONE.CRITICAL
            })
          })
        }
      ]
    }
  ]
};
