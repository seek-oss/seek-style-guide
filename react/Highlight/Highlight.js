// @flow
import styles from './Highlight.less';
import React from 'react';
import classnames from 'classnames';
import { TONE } from '../private/tone';

type Props = {
  children: React$Node,
  secondary?: boolean,
  className?: string,
  tone?: typeof TONE.CRITICAL
};

export default function Highlight(props: Props) {
  const { children, secondary, tone, className, ...restProps } = props;

  return (
    <mark
      {...restProps}
      className={classnames({
        [styles.root]: true,
        [styles.secondaryText]: secondary,
        [styles[tone]]: tone,
        ...(className ? { [className]: className } : {})
      })}
    >
      {children}
    </mark>
  );
}
