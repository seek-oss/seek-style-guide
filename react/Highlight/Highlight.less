@import (reference) "~seek-style-guide/theme";

.root {
  border-radius: 2px;
  display: inline-block;
  margin: -2px;
  padding: 2px;
  .highlightBackground(@sk-focus);
}

.highlightBackground(@color) {
  background-color: fade(@color, 12%);
}

.secondaryText {
  color: @sk-charcoal;
}

.critical {
  .highlightBackground(@sk-critical);
}
