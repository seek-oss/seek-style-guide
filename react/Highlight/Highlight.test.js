import React from 'react';
import { shallow } from 'enzyme';

import Highlight from './Highlight';

describe('Highlight:', () => {
  it('should render with defaults', () => {
    expect(shallow(<Highlight>text</Highlight>)).toMatchSnapshot();
  });

  it('should render with className', () => {
    expect(
      shallow(<Highlight className="foo">text</Highlight>)
    ).toMatchSnapshot();
  });

  it('should render with tone', () => {
    expect(
      shallow(
        <Highlight tone="critical" className="foo">
          text
        </Highlight>
      )
    ).toMatchSnapshot();
  });

  it('should render as secondary', () => {
    expect(
      shallow(
        <Highlight secondary tone="critical" className="foo">
          text
        </Highlight>
      )
    ).toMatchSnapshot();
  });
});
