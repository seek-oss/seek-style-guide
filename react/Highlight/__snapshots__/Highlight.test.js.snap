// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight: should render as secondary 1`] = `
<mark
  className="Highlight__root Highlight__secondaryText Highlight__critical foo"
>
  text
</mark>
`;

exports[`Highlight: should render with className 1`] = `
<mark
  className="Highlight__root foo"
>
  text
</mark>
`;

exports[`Highlight: should render with defaults 1`] = `
<mark
  className="Highlight__root"
>
  text
</mark>
`;

exports[`Highlight: should render with tone 1`] = `
<mark
  className="Highlight__root Highlight__critical foo"
>
  text
</mark>
`;
